indigo/indigo-extras/src/main/scala/indigoextras/pathfinding/PathFinder.scala

```diff
@@ -0,0 +1,127 @@
+package indigoextras.pathfinding
+
+import scala.annotation.tailrec
+
+/** The structure allowing to customize the path finding and to build a path of type T
+  *
+  * @tparam T
+  *   the type of the points
+  */
+trait PathBuilder[T]:
+  def neighbours(
+      t: T
+  ): List[T] // neighbours retrieval allows to select the allowed moves (horizontal, vertical, diagonal, impossible moves, jumps, etc.)
+  def distance(t1: T, t2: T): Int // distance allows to select the cost of each move (diagonal, slow terrain, etc.)
+  def heuristic(
+      t1: T,
+      t2: T
+  ): Int // heuristic allows to select the way to estimate the distance from a point to the end
+
+// A* (A Star) inspired algorithm version allowing generic types and customisation of the path finding
+object PathFinder:
+
+  private type OpenList[T]   = List[PathProps[T]] // the open list is the list of the points to explore
+  private type ClosedList[T] = List[PathProps[T]] // the closed list is the list of the points already explored
+
+  /** The structure containing the properties of a point
+    * @param value
+    *   the underlying value (coordinate, point, etc.)
+    * @param g
+    *   the distance from start
+    * @param h
+    *   the heuristic (distance from end)
+    * @param f
+    *   the g + h
+    * @param parent
+    *   the parent point
+    * @tparam T
+    *   the type of the points
+    */
+  private case class PathProps[T](value: T, g: Int, h: Int, f: Int, parent: Option[PathProps[T]] = None)
```

Comment:
Avoid default args, I'd rather be explicit everywhere, almost especially on internal plumbing.